app/helpers/vacancies_helper.rb

```diff
@@ -59,4 +59,11 @@ def page_title_prefix(vacancy, form_object, page_heading)
         'Error: ' : ''}#{page_heading} — #{t('jobs.create_a_job', school: current_school.name)}"
     end
   end
+
+  def missing_subjects?(vacancy)
+    legacy_subjects = [vacancy.subject,
+                       vacancy.first_supporting_subject,
+                       vacancy.second_supporting_subject].reject(&:blank?)
```

Comment:
It's much of a muchness, but I discovered the ruby method `.compact` yesterday, which rejects nil elements from an array. (`.blank?` captures a few more things that are unlikely to crop up here, e.g. `false`, and empty arrays.)